pkg/api/sharedresource/v1alpha1/share_type.go

```diff
@@ -13,58 +8,120 @@ import (
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Share is the Schema for the shares API
-type Share struct {
+// SharedResource allows a backing Secret or ConfigMap to be shared across namespaces.
+// Pods can mount the shared Secret or ConfigMap by adding a CSI volume to the pod specification using the
+// "sharedresource.storage.openshift.io" CSI driver and a reference to the SharedResource in the volume attributes:
+//
+// spec:
+//  volumes:
+//  - name: shared-secret
+//    csi:
+//      driver: sharedresource.storage.openshift.io
+//      volumeAttributes:
+//        sharedResource: my-share
+//
+// For the mount to be successful, the pod's service account must be granted permission to 'use' the named SharedResource object
+// within its namespace with an appropriate Role and RoleBinding.  For compactness, here are example `oc` invocations for creating
+// such Role and RoleBinding objects.
+//
+//  `oc create role shared-resource-my-share --verb=use --resource=sharedresources.storage.openshift.io --resource-name=my-share`
+//  `oc create rolebinding shared-resource-my-share --role=shared-resource-my-share --serviceaccount=my-namespace:default`
+//
+// Administrators can create separate Roles and RoleBindings for their users to be able the list and/or view the
+// available cluster scoped `SharedResources` objects.
```

Comment:
this is the end of where I tried to update godoc @coreydaley in reference to discussion with @deads2k in slack

ideally you can replicate this in your openshift/api PR, and then, with the resolution of the group name that @adambkaplan will be striving for with @jsafrane and @deads2k your openshift/api PR will be "unblocked", or at least at the point where we are just iterating on grammar and word choice vs. major concepts and architectural viability